cherrypy/lib/static.py

```diff
@@ -28,6 +29,21 @@ def _setup_mimetypes():
 _setup_mimetypes()
 
 
+def _utf8_content_disposition(disposition, file_name):
+    """Create HTTP header for downloading a file with UTF-8 filename.
+    See this and related answers:
+    https://stackoverflow.com/a/8996249/2173868
+    """
+    ascii_name = (
+        unicodedata.normalize('NFKD', file_name).
```

Comment:
I think this needs a code comment explaining why specifically `NKFD` is used here and not some other normalization type.

Reply:
There is good description of basic differences on wiki:  https://en.wikipedia.org/wiki/Unicode_equivalence

https://docs.python.org/3.7/library/unicodedata.html

__NFC__: Normal form C (NFC) first applies a canonical decomposition, then composes pre-combined characters again.
vs.
__NFKC__: The normal form KC (NFKC) first applies the compatibility decomposition, followed by the canonical composition.
vs.
__NFD__: Normal form D (NFD) is also known as canonical decomposition, and translates each character into its decomposed form.
vs.
__NFKD__: The normal form KD (NFKD) will apply the compatibility decomposition, i.e. replace all compatibility characters with their equivalents.

= ???

It looks like better solution to use __NKFC__ finally, because of what they say on wiki "_Sequences that are canonically equivalent are also compatible, but the opposite is not necessarily true._". What do you think?

Reply:
Okay, `NFKC` it is. But make sure to add explanation and references as `# code comments`.